onnxruntime/contrib_ops/cpu/tokenizer.h

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/framework/op_kernel.h"
+
+namespace onnxruntime {
+namespace contrib {
+class Tokenizer : public OpKernel {
+ public:
+  explicit Tokenizer(const OpKernelInfo& info);
+  Tokenizer(const Tokenizer&) = delete;
+  Tokenizer& operator=(const Tokenizer&) = delete;
+  ~Tokenizer();
+
+  Status Compute(OpKernelContext* context) const override;
+
+ private:
+  Status CharTokenize(OpKernelContext* context, size_t N, size_t C,
+                      const std::vector<int64_t>& input_dims) const;
+
+  Status SeparatorTokenize(OpKernelContext* context, size_t N, size_t C,
+                           const std::vector<int64_t>& input_dims) const;
+
+  bool mark_;
+  std::string pad_value_;
+  int64_t mincharnum_;
+  bool char_tokenezation_;
+  struct SearchData;
+  SearchData* search_data_;
```

Comment:
If you use a unique_ptr here, you can mark the destructor default.

Reply:
Not sure about it. SearchData definition should be available at the time of destruction and thus I'd like to explicitly put the destruction into .cc

---
In reply to: [238892480](https://github.com/Microsoft/onnxruntime/pull/31#discussion_r238892480) [](ancestors = 238892480)

Reply:
Sorry, didn't follow your comment. Why would search_data_ not be available at the time of destruction? Isn't it part of the class?

Reply:
With __dtor default it may be made inline and SearchData is forward declared only.

---
In reply to: [238894318](https://github.com/Microsoft/onnxruntime/pull/31#discussion_r238894318) [](ancestors = 238894318)

Reply:
Defining an empty destructor wouldn't do? Just trying to get rid of explicit 'new' and 'delete' from the source code as much as possible.

Reply:
Although it is a noble endeavor, delete is a fair game in the dtor. I am reading the site you mentioned and it is very opinionated, in my opinion.

---
In reply to: [238898621](https://github.com/Microsoft/onnxruntime/pull/31#discussion_r238898621) [](ancestors = 238898621)

Reply:
Why bother remembering to call delete if the same can be achieved using a unique_ptr which comes at pretty much no cost? Isn't it better to remember one less thing?

Reply:
I will correct it. :) 

---
In reply to: [238899988](https://github.com/Microsoft/onnxruntime/pull/31#discussion_r238899988) [](ancestors = 238899988)

Reply:
Thanks!